### PR TITLE
Replace HeapFree with CoTaskMemAlloc 

### PR DESCRIPTION
**Pull request by timsneath**
per https://github.com/timsneath/win32/issues/443#issuecomment-1139768737.

This works locally with my win32 project identically to the HeapAlloc version.

### CODE REVIEW

File: lib/src/allocation.dart [hunk 1]
```diff
@@ -25,39 +25,31 @@ typedef PosixFree = void Function(Pointer);
 final PosixFree posixFree =
     stdlib.lookupFunction<PosixFreeNative, PosixFree>('free');
 
-typedef WinGetProcessHeapFn = Pointer Function();
-final WinGetProcessHeapFn winGetProcessHeap = stdlib
-    .lookupFunction<WinGetProcessHeapFn, WinGetProcessHeapFn>('GetProcessHeap');
-final Pointer processHeap = winGetProcessHeap();
+typedef WinCoTaskMemAllocNative = Pointer Function(Size cb);
```
Comment (dcharkes) [inline lib/src/allocation.dart:28]: Size has been introduced to the Dart SDK in 2.17, so we need to bump the minimum SDK version to 2.17 and update the bot testing stable to 2.17.

Comment (dcharkes) [inline lib/src/allocation.dart:28]: Size has been introduced to the Dart SDK in 2.17, so we need to bump the minimum SDK version to 2.17 and update the bot testing stable to 2.17.

File: lib/src/allocation.dart [hunk 2]
```diff
@@ -25,39 +25,31 @@ typedef PosixFree = void Function(Pointer);
...
+    .lookupFunction<WinCoTaskMemFreeNative, WinCoTaskMemFree>('CoTaskMemFree');
 
 /// Manages memory on the native heap.
 ///
 /// Does not initialize newly allocated memory to zero. Use [_CallocAllocator]
 /// for zero-initialized memory on allocation.
 ///
 /// For POSIX-based systems, this uses `malloc` and `free`. On Windows, it uses
-/// `HeapAlloc` and `HeapFree` against the default public heap.
+/// `CoTaskMemAlloc`.
```
Comment (dcharkes) [inline lib/src/allocation.dart:45]: Now that we change the underlying implementation, we break people if they were mixing and matching `package:ffi` and their own calls to `HeapAlloc` and `HeapFree`.

Let's bump `package:ffi` to 2.0.0.

**Review by dcharkes — APPROVED** [review]
Sweet @timsneath! 😄 

Let's bump package:ffi to 2.0.0 as this is potentially a breaking change for people.

And we need to bump the SDK to 2.17 to use the `Size` type.

lgtm

**Comment by timsneath** [comment]
Great! I bumped those changes. I also updated the `dev_dependencies` to point to the latest versions of lints and tests.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,21 +1,26 @@
 # Changelog
 
+## 2.0.0
+
+Switch Windows memory allocation to use `CoTaskMemAlloc` and `CoTaskMemFree`,
+which will enable support for `NativeFinalizer`.
+
```
Comment (dcharkes) [inline CHANGELOG.md:7]: Nit: This release requires Dart `2.17.0` or greater.

Comment (timsneath) [reply CHANGELOG.md:8]: Great! Updated.

**Comment by dcharkes** [comment]
Thank you very much good sir!

**Pull request merged**